micro-rdk/src/common/registry.rs

```diff
@@ -152,6 +165,8 @@ impl Default for ComponentRegistry {
             crate::common::generic::register_models(&mut r);
             crate::common::ina::register_models(&mut r);
             crate::common::wheeled_base::register_models(&mut r);
+            #[cfg(feature = "camera")]
```

Comment:
`fake_camera` only exists if the "native" feature is activated, but here we attempt to register models even if the native feature is not active. Your PR seems to be passing CI checks despite this but I'm not sure how?

EDIT: it's because we don't pass the camera feature in any of our CI checks (which I'm not necessarily advocating for doing). I've confirmed that the following fails
```
cargo +esp clippy -p micro-rdk  --features esp32,provisioning,camera  --target=xtensa-esp32-espidf -Zbuild-std=std,panic_abort -- -Dwarnings
```
The follow-up question here is: should we seek a more graceful compilation failure via [compile_error](https://doc.rust-lang.org/std/macro.compile_error.html)?

Reply:
For the `Esp32Camera`, this ticket only covers native. In the [epic](https://viam.atlassian.net/browse/RSDK-7813?atlOrigin=eyJpIjoiYzFmMWNkNmZjNzkxNDVmMjgxNzY0MDNiNTllZWMyMDIiLCJwIjoiaiJ9), esp32 implementations and testings come later.

Reply:
I can do an immediate PR after so it passes for esp32,camera. Or if you don't mind waiting a week and I fix it when I get to the esp32 stuff

Reply:
Could you put the native flag above this line though? Otherwise even when you fix `Esp32Camera` it will still fail